### PR TITLE
Feature: Allow for a single object upload for posts 

### LINKED ISSUE #1127: Allow for a single object upload for posts
**Is your feature request related to a problem? Please describe.**
In the create post modal we have 2 different fields to upload video and image post.

**Describe the solution you'd like**
we want to make  a single field that allows posting both images and videos through it, you can name the field as "Add media".

**Additional context**
make sure you manage the preview of the content within the model itself and manage the close preview state.
Screenshot 2023-12-05 at 9 19 09 PM

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by duplixx**
**What kind of change does this PR introduce?**

**Issue Number:**

Fixes #1127 

**Did you add tests for your changes?**

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/e8c4ecdb-26eb-4d40-a834-d0de0c248f47

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by duplixx** [comment]
@aashimawadhwa, can you review it and let me know if any changes are required?

**Comment by aashimawadhwa** [comment]
can you fix the linting workflow? @duplixx

**Comment by palisadoes** [comment]
The linting config file has been reverted. Please merge with your upstream and make an updating commit.

**Comment by aashimawadhwa** [comment]
@duplixx  can you fix the failing test.

**Comment by duplixx** [comment]
I'm currently stuck in an error that doesn't make sense. Could you please take a look at it, @aashimawadhwa, @palisadoes, @noman2002? I'll be attaching a video for better understanding.
https://github.com/PalisadoesFoundation/talawa-admin/assets/90516956/8cac51cd-3faa-4234-b027-a18b5d326d5f

**Comment by duplixx** [comment]
@aashimawadhwa @palisadoes Great news! I fixed the error by running it on a different machine. Other than that, we may need to add more test cases in the future. For now, I've included the test cases for the code I added.

**Comment by duplixx** [comment]
hi @aashimawadhwa @EshaanAgg @noman2002 can you review this 
thanks!!

**Comment by Cioppolo14** [comment]
@duplixx  There are still failing tests.  Are you working on this or stuck?

**Comment by aashimawadhwa** [comment]
@duplixx  updates?

**Comment by duplixx** [comment]
Hi @aashimawadhwa, I am working in it to resolve the issue, but any help will be greatly appreciated.

**Comment by rishav-jha-mech** [comment]
@duplixx while running tests 

```sh
./coverage/lcov-report/index.html
```

run this file on live server, so that you can see which block of code is not being executed/covered by the tests. 

Also you while running tests you can use this command, 
```sh
npm run jest preview
```

to see how the UI changes, or what effects take place for a particular test. This is the standard way to write tests and debug in talawa-admin.

File: public/locales/en.json [hunk 1]
```diff
@@ -288,6 +288,7 @@
     "postDetails": "Post Details",
     "postTitle1": "Write title of the post",
     "postTitle": "Title",
+    "addMedia": "Upload Photo or Video",
```
Comment (rishav-jha-mech) [inline public/locales/en.json:291]: If you are adding any new text in the project, please make sure to add it for other languages too.

File: src/screens/OrgPost/OrgPost.tsx [hunk 2]
```diff
@@ -110,10 +114,10 @@ function orgPost(): JSX.Element {
...
           text: postinfo,
           organizationId: currentUrl,
-          file: postImage || postVideo,
+          file: postImage || postVideo || postformState.addMedia,
         },
       });
-      /* istanbul ignore next */
+
       if (data) {
         toast.success('Congratulations! You have Posted Something.');
```
Comment (rishav-jha-mech) [inline src/screens/OrgPost/OrgPost.tsx:122]: You can add a multilingual message here.

File: public/locales/en.json (hunk 1 above)
Comment (duplixx) [reply public/locales/en.json:291]: Thanks @rishav-jha-mech, i'll add it in different languages🙌

**Comment by duplixx** [comment]
Hey @rishav-jha-mech, I have added multilingual support for text rendering. I'm facing difficulty in implementing test cases for toasters. If you have any hints on how to add test cases for that, it would be helpful.

**Comment by rishav-jha-mech** [comment]
@duplixx you need to create a mock fn for Toast, go to this file and see how it is implemented, you will get the idea.
 `src\components\OrgPostCard\OrgPostCard.test.tsx`

**Comment by duplixx** [comment]
image
image

image


Hi @rishav-jha-mech 
I was adding a test case for this line, and the test case I added looks fine, but when I tried running it, it showed an error.

image

**Comment by duplixx** [comment]
Hi @rishav-jha-mech,
I've implemented the functionality and made the necessary changes. We can open an issue later to ensure complete test coverage of 100%.

**Comment by rishav-jha-mech** [comment]
@duplixx why do you wanna merge into main ?, set the base branch to develop and fix the tests

**Comment by duplixx** [comment]
@rishav-jha-mech should i open pr in another branch to fix this ??

**Comment by rishav-jha-mech** [comment]
@duplixx you can open a new PR for this, keep this PR as a reference to the new one.

**Comment by noman2002** [comment]
@duplixx Should this be closed ??

**Comment by duplixx** [comment]
yeah sure @noman2002

**Pull request closed without merging**